Packet++/src/DnsResourceData.cpp

```diff
@@ -96,7 +96,7 @@ bool IPv6DnsResourceData::toByteArr(uint8_t* arr, size_t& arrLength, IDnsResourc
 	return true;
 }
 
-MxDnsResourceData::MxDnsResourceData(uint8_t* dataPtr, size_t dataLen, IDnsResource* dnsResource)
+MxDnsResourceData::MxDnsResourceData(uint8_t* dataPtr, size_t, IDnsResource* dnsResource)
```

Comment:
Ditto I think that we should check that the preference is not ouf of bound and then pass dataLen - sizeof(preference) to decodename()